modules/space/rendering/renderableconstellationlines.cpp

```diff
@@ -0,0 +1,449 @@
+    constexpr openspace::properties::Property::PropertyInfo DrawElementsInfo = {
+        "DrawElements",
+        "Draw Elements",
+        "Enables/Disables the drawing of the constellations"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo ConstellationUnitInfo = {
+        "ConstellationUnit",
+        "Constellation Unit",
+        "The unit used for the constellation data"
+    };
+
+    constexpr openspace::properties::Property::PropertyInfo ConstellationColorInfo = {
+        "ConstellationColor",
+        "Constellation colors",
+        "The defined colors for the constellations to be rendered"
+    };
+
+    struct [[codegen::Dictionary(RenderableConstellationLines)]] Parameters {
+        // The path to the SPECK file that contains constellation lines data
+        std::filesystem::path file;
+
+        enum class [[codegen::map(openspace::DistanceUnit)]] Unit {
+            Meter [[codegen::key("m")]],
+            Kilometer [[codegen::key("Km")]],
+            Parsec [[codegen::key("pc")]],
+            Kiloparsec [[codegen::key("Kpc")]],
+            Megaparsec [[codegen::key("Mpc")]],
+            Gigaparsec [[codegen::key("Gpc")]],
+            Gigalightyear [[codegen::key("Gly")]]
+        };
+        // [[codegen::verbatim(ConstellationUnitInfo.description)]]
+        std::optional<Unit> constellationUnit;
+
+        // [[codegen::verbatim(ConstellationColorInfo.description)]]
+        std::optional<std::vector<glm::vec3>> constellationColor;
+    };
+#include "renderableconstellationlines_codegen.cpp"
+}  // namespace
+
+namespace openspace {
+
+documentation::Documentation RenderableConstellationLines::Documentation() {
+    return codegen::doc<Parameters>("space_renderable_constellationlines");
+}
+
+RenderableConstellationLines::RenderableConstellationLines(
+                                                      const ghoul::Dictionary& dictionary)
+    : RenderableConstellation(dictionary)
+    , _drawElements(DrawElementsInfo, true)
+{
+    const Parameters p = codegen::bake<Parameters>(dictionary);
+
+    _speckFile = absPath(p.file.string()).string();
+    _hasSpeckFile = true;
+    _drawElements.onChange([&]() { _hasSpeckFile = !_hasSpeckFile; });
+    addProperty(_drawElements);
+
+    if (p.constellationUnit.has_value()) {
+        _constellationUnit = codegen::map<DistanceUnit>(*p.constellationUnit);
+    }
+    else {
+        _constellationUnit = DistanceUnit::Meter;
+    }
+
+    if (p.constellationColor.has_value()) {
+        std::vector<glm::vec3> ops = *p.constellationColor;
+        for (size_t i = 0; i < ops.size(); ++i) {
+            _constellationColorMap.insert({ static_cast<int>(i) + 1, ops[i] });
+        }
+    }
+}
+
+void RenderableConstellationLines::selectionPropertyHasChanged() {
+    // If no values are selected (the default), we want to show all constellations
+    if (!_constellationSelection.hasSelected()) {
+        for (std::pair<const int, ConstellationLine>& pair :
```

Comment:
Maybe make named a type for `std::pair<const int, ConstellationLine>`, to make it clearer what it is and easier to read? :)